packages/plugins/typescript/resolvers/tests/mapping.spec.ts

```diff
@@ -7,23 +7,48 @@ describe('ResolversTypes', () => {
   it('Should build ResolversTypes object when there are no mappers', async () => {
     const result = await plugin(resolversTestingSchema, [], {}, { outputFile: '' });
 
+    expect(result.content).toBeSimilarStringTo(`
+      export type ResolversUnionTypes = {
+        ChildUnion: ( Child ) | ( MyOtherType );
+        MyUnion: ( Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversTypes['ChildUnion']> } ) | ( MyOtherType );
+      };
+    `);
     expect(result.content).toBeSimilarStringTo(`
     export type ResolversTypes = {
       MyType: ResolverTypeWrapper<Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversTypes['ChildUnion']> }>;
       String: ResolverTypeWrapper<Scalars['String']>;
       Child: ResolverTypeWrapper<Child>;
       MyOtherType: ResolverTypeWrapper<MyOtherType>;
-      ChildUnion: ResolversTypes['Child'] | ResolversTypes['MyOtherType'];
+      ChildUnion: ResolverTypeWrapper<ResolversUnionTypes['ChildUnion']>;
       Query: ResolverTypeWrapper<{}>;
       Subscription: ResolverTypeWrapper<{}>;
       Node: ResolversTypes['SomeNode'];
       ID: ResolverTypeWrapper<Scalars['ID']>;
       SomeNode: ResolverTypeWrapper<SomeNode>;
-      MyUnion: ResolversTypes['MyType'] | ResolversTypes['MyOtherType'];
+      MyUnion: ResolverTypeWrapper<ResolversUnionTypes['MyUnion']>;
       MyScalar: ResolverTypeWrapper<Scalars['MyScalar']>;
       Int: ResolverTypeWrapper<Scalars['Int']>;
       Boolean: ResolverTypeWrapper<Scalars['Boolean']>;
-    };`);
+    };
+    `);
+    expect(result.content).toBeSimilarStringTo(`
+      export type ResolversParentTypes = {
+        MyType: Omit<MyType, 'unionChild'> & { unionChild?: Maybe<ResolversParentTypes['ChildUnion']> };
+        String: Scalars['String'];
+        Child: Child;
+        MyOtherType: MyOtherType;
+        ChildUnion: ResolversUnionTypes['ChildUnion'];
+        Query: {};
+        Subscription: {};
+        Node: ResolversParentTypes['SomeNode'];
+        ID: Scalars['ID'];
+        SomeNode: SomeNode;
+        MyUnion: ResolversUnionTypes['MyUnion'];
+        MyScalar: Scalars['MyScalar'];
+        Int: Scalars['Int'];
+        Boolean: Scalars['Boolean'];
+      };
+    `);
```

Comment:
Added `ResolversParentTypes` to all related tests in this file because I want to be sure that `ResolversUnionTypes` is being used correctly.